src/main/java/picard/illumina/parser/NewIlluminaDataProvider.java

```diff
@@ -66,7 +69,7 @@
         //barcodes
         final Pattern barcodeRegex = Pattern.compile(ParameterizedFileUtil.escapePeriods(
                 ParameterizedFileUtil.makeBarcodeRegex(lane)));
-        final File[] barcodeFiles = getTiledFiles(basecallDirectory, barcodeRegex);
+        final File[] barcodeFiles = getTiledFiles(barcodesDirectory, barcodeRegex);
```

Comment:
Would it make sense to throw an exception here if barcodeFiles is empty?
Also add a ` IOUtil.assertFilesAreReadable(Arrays.asList(barcodeFiles));'
as is done for the locs and filter files.

Reply:
Done